examples/default.yaml

```diff
@@ -27,9 +27,23 @@ images:
 - location: "https://cloud-images.ubuntu.com/releases/24.04/release-20240725/ubuntu-24.04-server-cloudimg-amd64.img"
   arch: "x86_64"
   digest: "sha256:d2377667ea95222330ca2287817403c85178dad397e9fed768a9b4aec79d2a7f"
+  kernel:
+    location: "https://cloud-images.ubuntu.com/releases/24.04/release-20240725/unpacked/ubuntu-24.04-server-cloudimg-amd64-vmlinuz-generic"
+    digest: "sha256:47bf2de2f093931bcb8e862b3390905038de301a9e9a8bc1b462666bb4011dcd"
+    cmdline: "root=LABEL=cloudimg-rootfs ro console=tty1 console=ttyAMA0 no_timer_check"
```

Comment:
I think this is fine for CI, however, this makes impossible to update the kernel and initrd via apt-get, so I guess we shouldn't do this outside the CI. (The panic issue doesn't seem to happen outside the CI AFAICS.)

Reply:
Ideally, Lima should be able to mount the qcow2 image and patch the grub config inside it, but its implementation cost is likely to be extremely high.

Reply:
I also wonder if the kernel cmdline could be supplied via an EFI var, fw_cfg, ACPI table, etc.

Reply:
> I also wonder if the kernel cmdline could be supplied via an EFI var, fw_cfg, ACPI table, etc.

I looked into the methods you suggested, but it seems quite complicated. Instead, how about adding an option in `lima.yaml` that only uses kernel-related settings during the initial boot, such as `.kernel.initial_boot_only`, `.initial_boot_kernel`, `.initial_boot_initrd`?

Reply:
Sounds overengineering.
On the CI, `no_timer_check` is needed on every boot of the instance, and it is not needed outside the CI.

Reply:
I guess CI should just use `yq` to inject the cmdline

Reply:
Updated to inject kernel related settings to template.